include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,648 @@
+    */
+    size_type
+    size() const noexcept
+    {
+        return dn_;
+    }
+
+    /** Returns whether the string is empty
+    */
+    bool
+    empty() const noexcept
+    {
+        return n_ == 0;
+    }
+
+    /** Copies a decoded substring to another character string
+
+        Copies a substring to the character array pointed to by
+        the destination char string, where `rcount` is the smaller
+        of `count` and `size() - pos`.
+
+        @par Exception Safety
+        Strong guarantee.
+        Exceptions thrown on invalid positions.
+
+        @throw std::out_of_range If `pos > size()`
+
+        @param dest pointer to the destination character string
+        @param count requested substring length
+        @param pos position of the first character
+
+        @return Number of characters copied
+    */
+    BOOST_URL_DECL
+    size_type
+    copy(char* dest,
+         size_type count,
+         size_type pos = 0) const;
+
+    /** Return true if the corresponding comparison holds
+
+        The length of the sequences to compare is the smaller of
+        `size()` and `other.size()`.
+
+        The function compares the two strings as if by calling
+        `char_traits<char>::compare(to_string().data(), v.data(), rlen)`.
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_URL_DECL
+    int
+    compare(string_view other) const noexcept;
+
+    /// @copydoc compare()
+    int
+    compare(char const* other) const noexcept
+    {
+        return compare(string_view(other));
+    }
+
+    /** Return true if the corresponding comparison holds
+
+        The length of the sequences to compare is the smaller of
+        `size()` and `other.size()`.
+
+        The function compares the two strings as if by calling
+        `char_traits<char>::compare(to_string().data(), v.to_string().data(), rlen)`.
+
+        @param other string to compare
+
+        @return Negative value if this string is less than the other
+        character sequence, zero if the both character sequences are
+        equal, positive value if this string is greater than the other
+        character sequence
+    */
+    BOOST_URL_DECL
+    int
+    compare(decoded_view other) const noexcept;
+
+    /** Returns a string with the decoded data
+    */
+    std::string
+    to_string() const
+    {
+        std::string r;
+        assign_to(r);
+        return r;
+    }
+
+    /** Assigns the decoded characters to a container
+
+        This overload allows the caller to recycle
+        the input string.
+
+        This function only participates in
+        overload resolution if the container
+        type `T` has the member function
+        `assign(iterator, iterator)`.
```

Comment:
This is not correct. The function should always be found (no SFINAE) and we should put a BOOST_STATIC_ASSERT in the definition instead. This helps the user with better diagnostics, and you can put a comment near the assert explaining the requiremetns.